### PR TITLE
JAMES-342{3-4} JWT: security and documentation

### PR DESCRIPTION
**Pull request by chibenwa**
- JMAP and webadmin should use distinct signing keys

 - Better document JWT usage

### CODE REVIEW

File: docs/modules/servers/pages/distributed/configure/jmap.adoc [hunk 1]
```diff
@@ -36,3 +36,25 @@ This should not be the same keystore than the ones used by TLS based protocols.
 Enabling *TRACE* on `org.apache.james.jmap.wire` enables reactor-netty wiretap, logging of
 all incoming and outgoing requests, outgoing requests. This will log also potentially sensible information
 like authentication credentials.
+
+== Generqting a JWT key pair
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/jmap.adoc:40]:
```suggestion
== Generating a JWT key pair
```
Writing this on a qwerty? :trollface:

File: docs/modules/servers/pages/distributed/configure/jmap.adoc [hunk 2]
```diff
@@ -36,3 +36,25 @@ This should not be the same keystore than the ones used by TLS based protocols.
...
+
+The Distributed server enforces the use of RSA-SHA-256.
+
+One can use OpenSSL to generate a JWT tkey pair :
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/jmap.adoc:44]:
```suggestion
One can use OpenSSL to generate a JWT key pair :
```

File: docs/modules/servers/pages/distributed/configure/jmap.adoc [hunk 3]
```diff
@@ -36,3 +36,25 @@ This should not be the same keystore than the ones used by TLS based protocols.
...
+    openssl genrsa -out rs256-4096-private.rsa 4096
+    # public key
+    openssl rsa -in rs256-4096-private.rsa -pubout > rs256-4096-public.pem
+
+The private key can be used to generate JWT tokens, for instance
+using link:https://github.com/vandium-io/jwtgen[jwtgen]:
+
+    jwtgen -a RS256 -p rs256-4096-private.rsa 4096 -c "sub=bob@domain.tld" -e 3600 -V
+
+This token can then be passed as `Bearrer` of the `Authorization` header :
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/jmap.adoc:56]:
```suggestion
This token can then be passed as `Bearer` of the `Authorization` header :
```

File: docs/modules/servers/pages/distributed/configure/webadmin.adoc [hunk 4]
```diff
@@ -64,3 +64,25 @@ needs to be on the classpath or in the ./extensions-jars folder. Read mode about
 xref:distributed/extending/webadmin-routes.adoc[creating you own webadmin routes].
 
 |===
+
+== Generqting a JWT key pair
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/webadmin.adoc:68]:
```suggestion
== Generating a JWT key pair
```

File: docs/modules/servers/pages/distributed/configure/webadmin.adoc [hunk 5]
```diff
@@ -64,3 +64,25 @@ needs to be on the classpath or in the ./extensions-jars folder. Read mode about
...
+
+The Distributed server enforces the use of RSA-SHA-256.
+
+One can use OpenSSL to generate a JWT tkey pair :
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/webadmin.adoc:72]:
```suggestion
One can use OpenSSL to generate a JWT key pair :
```

File: docs/modules/servers/pages/distributed/configure/webadmin.adoc [hunk 6]
```diff
@@ -64,3 +64,25 @@ needs to be on the classpath or in the ./extensions-jars folder. Read mode about
...
+    openssl genrsa -out rs256-4096-private.rsa 4096
+    # public key
+    openssl rsa -in rs256-4096-private.rsa -pubout > rs256-4096-public.pem
+
+The private key can be used to generate JWT tokens, for instance
+using link:https://github.com/vandium-io/jwtgen[jwtgen]:
+
+    jwtgen -a RS256 -p rs256-4096-private.rsa 4096 -c "sub=bob@domain.tld" -e 3600 -V
+
+This token can then be passed as `Bearrer` of the `Authorization` header :
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/configure/webadmin.adoc:84]:
```suggestion
This token can then be passed as `Bearer` of the `Authorization` header :
```

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Some wiring missing with JPA it seems

```
[f7afc31cfcdfe26f8c358e49471b579e6851805b] [ERROR] Errors: 
[f7afc31cfcdfe26f8c358e49471b579e6851805b] [ERROR]   JPAJamesServerTest » Creation Unable to create injector, see the following err...
[f7afc31cfcdfe26f8c358e49471b579e6851805b] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest » Creation Unable to ...
[f7afc31cfcdfe26f8c358e49471b579e6851805b] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest » Creation Una...
[f7afc31cfcdfe26f8c358e49471b579e6851805b] [ERROR]   JamesCapabilitiesServerTest.startShouldSucceedWhenRequiredCapabilities » Creation
```

File: docs/modules/servers/pages/distributed/configure/jmap.adoc (hunk 1 above)
Comment (chibenwa) [reply docs/modules/servers/pages/distributed/configure/jmap.adoc:40]: Yes

**Review by Arsnael — APPROVED** [review]

**Review by nvduc91 — APPROVED** [review]

**Comment by chibenwa** [comment]
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project james-server-task-distributed: There are test failures.`

```
[0261b796e226eb9e8eadaaff0d2baa5e95ece409] [ERROR]   DistributedTaskManagerTest.badPayloadsShouldNotAffectTaskManagerOnCancelTask:258 
[0261b796e226eb9e8eadaaff0d2baa5e95ece409] Expecting:
[0261b796e226eb9e8eadaaff0d2baa5e95ece409]  <COMPLETED>
[0261b796e226eb9e8eadaaff0d2baa5e95ece409] to be equal to:
[0261b796e226eb9e8eadaaff0d2baa5e95ece409]  <CANCELLED>
[0261b796e226eb9e8eadaaff0d2baa5e95ece409] but was not.
```

test this please

**Comment by Arsnael** [comment]
```
[f4c024af545fa2bf689c2150e9a154659549fcfd] [ERROR] Failures: 
[f4c024af545fa2bf689c2150e9a154659549fcfd] [ERROR]   RabbitMQJwtFilterIntegrationTest>JwtFilterIntegrationTest.jwtAuthenticationShouldWork:82 1 expectation failed.
```
related maybe?

**Comment by chibenwa** [comment]
Maybe...

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**